modin/experimental/core/execution/native/implementations/hdk_on_native/expr.py

```diff
@@ -944,7 +968,46 @@ def copy(self):
         -------
         OpExpr
         """
-        return OpExpr(self.op, self.operands.copy(), self._dtype)
+        op = OpExpr(self.op, self.operands.copy(), self._dtype)
+        if pk := getattr(self, "partition_keys", None):
+            op.partition_keys = pk
+            op.is_rows = self.is_rows
+            op.order_keys = self.order_keys
+            op.lower_bound = self.lower_bound
+            op.upper_bound = self.upper_bound
+        return op
+
+    @_inherit_docstrings(BaseExpr.nested_expressions)
+    def nested_expressions(
+        self,
+    ) -> Generator[Type["BaseExpr"], Type["BaseExpr"], Type["OpExpr"]]:
+        expr = self
+        for i, op in enumerate(self.operands):
+            new_op = yield op
+            if new_op is not None:
+                if new_op is not op:
+                    if expr is self:
+                        expr = self.copy()
+                    expr.operands[i] = new_op
+                yield expr
```

Comment:
At least in two places there is a generator by `operands`, can it be taken out into a separate function?